tests/boards/intel_adsp/hda/src/smoke.c

```diff
@@ -0,0 +1,189 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <ztest.h>
+#include <cavs_ipc.h>
+#include <cavs_hda.h>
+#include "tests.h"
+
+#define IPC_TIMEOUT K_MSEC(100)
+#define STREAM_ID 3U
+#define FIFO_SIZE 256
+#define TRANSFER_SIZE 256
+#define TRANSFER_COUNT 8
+
+__attribute__((section(".dma_buffers"), aligned(128))) uint8_t hda_fifo[FIFO_SIZE];
+
+
+static volatile int msg_cnt;
+static volatile int msg_res;
+
+static bool ipc_message(const struct device *dev, void *arg,
+			uint32_t data, uint32_t ext_data)
+{
+	printk("HDA message received, data %u, ext_data %u\n", data, ext_data);
+	msg_res = data;
+	msg_cnt++;
+	return true;
+}
+
+static void ipc_done(const struct device *dev, void *arg)
```

Comment:
It should work to register a null handler, you don't have to have an empty function.

Reply:
I had some debug printks in here before, removable now. Noted for the cleanup round